blivet/tasks/fsuuid.py

```diff
@@ -0,0 +1,81 @@
+import abc
+
+from string import hexdigits
+from six import add_metaclass
+
+
+@add_metaclass(abc.ABCMeta)
+class FSUUID(object):
+
+    """An abstract class that represents filesystem actions for setting the
+       UUID.
+    """
+
+    @abc.abstractmethod
+    def uuid_format_ok(self, uuid):
+        """Returns True if the given UUID is correctly formatted for
+           this filesystem, otherwise False.
+
+           :param str uuid: the UUID for this filesystem
+           :rtype: bool
+        """
+        raise NotImplementedError
+
+    # IMPLEMENTATION methods
+
+    @classmethod
+    def _check_rfc4122_uuid(cls, uuid):
+        """Check whether the given UUID is correct according to RFC 4122 and
+           return True if it's correct or False otherwise.
+
+           :param str uuid: the UUID to check
+           :rtype: bool
+        """
+        chunks = uuid.split('-')
+        if len(chunks) != 5:
+            return False
+        chunklens = [len(chunk) for chunk in chunks
+                     if all(char in hexdigits for char in chunk)]
+        return chunklens == [8, 4, 4, 4, 12]
```

Comment:
Are these the only restrictions specified by the RFC? Or should this be using some regexp to limit the character sets as well instead?